fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneStoredFieldsTest.java

```diff
@@ -0,0 +1,395 @@
+import com.apple.foundationdb.record.query.plan.plans.RecordQueryPlan;
+import com.apple.foundationdb.tuple.Tuple;
+import com.apple.test.BooleanSource;
+import com.apple.test.Tags;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Message;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.params.ParameterizedTest;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.SIMPLE_TEXT_SUFFIXES;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.TEXT_AND_STORED_COMPLEX;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.createSimpleDocument;
+import static com.apple.foundationdb.record.metadata.Key.Expressions.concat;
+import static com.apple.foundationdb.record.metadata.Key.Expressions.field;
+import static com.apple.foundationdb.record.metadata.Key.Expressions.function;
+import static com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexTestUtils.COMPLEX_DOC;
+import static com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexTestUtils.SIMPLE_DOC;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests for Lucene StoredFields implementation.
+ */
+@Tag(Tags.RequiresFDB)
+public class LuceneStoredFieldsTest extends FDBRecordStoreTestBase {
+
+    // The index to use when teh optimized stored fields option is enabled (this is the same as the base index, with the
+    // PK index and Optimized stored fields options enabled)
+    private static final Index SIMPLE_TEXT_SUFFIXES_WITHOUT_OPT_STORED_FIELDS = new Index("Simple$text_suffixes_pky",
+            function(LuceneFunctionNames.LUCENE_TEXT, field("text")),
+            LuceneIndexTypes.LUCENE,
+            ImmutableMap.of(IndexOptions.TEXT_TOKENIZER_NAME_OPTION, AllSuffixesTextTokenizer.NAME,
+                    LuceneIndexOptions.OPTIMIZED_STORED_FIELDS_FORMAT_ENABLED, "false"));
+
+    public static final Index TEXT_AND_STORED_COMPLEX_WITHOUT_OPT_STORED_FIELDS = new Index(
+            "Simple$test_stored_complex",
+            concat(function(LuceneFunctionNames.LUCENE_TEXT, field("text")),
+                    function(LuceneFunctionNames.LUCENE_STORED, field("text2")),
+                    function(LuceneFunctionNames.LUCENE_STORED, field("group")),
+                    function(LuceneFunctionNames.LUCENE_STORED, field("score")),
+                    function(LuceneFunctionNames.LUCENE_STORED, field("time")),
+                    function(LuceneFunctionNames.LUCENE_STORED, field("is_seen"))),
+            LuceneIndexTypes.LUCENE,
+            ImmutableMap.of(LuceneIndexOptions.OPTIMIZED_STORED_FIELDS_FORMAT_ENABLED, "false"));
```

Comment:
should this enable the primarykey index?

Reply:
No - the PK index is enabled and disabled through the PK-index specific tests. Since the desired state is to have the stored fields option enabled, I opted to have the base index definition as `TRUE` so that all future tests are including it by default. The PK index option is defaulted to `FALSE` so the only way they are tested as both `TRUE` is in the PK index tests.
I think it may be worthwhile to invert the PK index option on the base indexes, later.